src/operator/control_flow.cc

```diff
@@ -1225,6 +1225,9 @@ static bool BackwardCondStorageType(const nnvm::NodeAttrs& attrs,
     CHECK(sync_in_in(input_locs, out_attrs, &subg_out_attrs, is_udf));
     return ret;
   };
+  for (const dim_t &cond_in : params.cond_input_locs) {
+    (*out_attrs)[cond_in] = 0;
+  }
```

Comment:
what is this for?

Reply:
`cond` don't need to be back-propagated

Reply:
Is 0 a magic number?

Reply:
@eric-haibin-lin In this case, anything other than `kBadStorageID` will work because we don't bp through `cond`. Do you suggest that I use `kDefaultStorage` to make it clearer?